nova/json.go

```diff
@@ -88,7 +88,17 @@ func (entity *Entity) UnmarshalJSON(b []byte) error {
 	var je jsonEntity = jsonEntity(*entity)
 	var err error
 	if err = json.Unmarshal(b, &je); err != nil {
-		return err
+		// Related Bug lp:1683495, within the openstack compute api
+		// response to List Servers Detailed, the image object might
+		// be an empty string when you boot the server from a volume.
+		// Before failing here, let's check to see if that's the case.
+		err2 := json.Unmarshal(b, &je.Id)
```

Comment:
Maybe just test `if string(b) == "\"\""`? Feels a little odd to unmarshal into a field when we know it can only be an empty string.